templates/etc/php/7.3/fpm/pool.d/www.conf.patch

```diff
@@ -0,0 +1,22 @@
+@@@ Configure php_fpm to use a larger, fixed number of threads
+--- debian/etc/php/7.3/fpm/pool.d/www.conf
++++ permanent/etc/php/7.3/fpm/pool.d/www.conf
+@@ -100,7 +100,7 @@
+ ;             pm.process_idle_timeout   - The number of seconds after which
+ ;                                         an idle process will be killed.
+ ; Note: This value is mandatory.
+-pm = dynamic
++pm = static
+
+ ; The number of child processes to be created when pm is set to 'static' and the
+ ; maximum number of child processes when pm is set to 'dynamic' or 'ondemand'.
+@@ -111,7 +111,7 @@
+ ; forget to tweak pm.* to fit your needs.
+ ; Note: Used when pm is set to 'static', 'dynamic' or 'ondemand'
+ ; Note: This value is mandatory.
+-pm.max_children = 5
++pm.max_children = 400
```

Comment:
Does this have to be the same number of children as Apache's MaxRequestWorkers, or is that coincidental?

Reply:
We (me and @meisekimiu ) think that we need some Apache workers to serve static web app assets, which will not go through PHP, as well, so this should be a bit smaller than the MaxRequestWorkers.

Reply:
I intuitively assumed they should be the same, but yeah let's talk it out. You're right that there are still static assets that need to be served (although hopefully fewer and fewer? what's left?).

I tested this in a controlled environment, and it appears that if Apache has more workers than PHP, then when there are more PHP requests than PHP workers, Apache will queue requests until PHP workers are available.

If PHP has more workers than Apache, then those extra PHP processes are wasted overhead.

In any case, if there are more requests than Apache workers, those requests will be queued, but if queued for too long may begin timing out.

So, there doesn't appear to be a significant risk in keeping them in sync with each other, but reserving some workers for static requests seems reasonable. How many would you propose?

Reply:
10%?

Reply:
Done, and rebased on `main`.